### PR TITLE
Wasm Binary Size Optimizations + Expansion of Wasm Exports

### PR DESCRIPTION
**Pull request by iamalwaysuncomfortable**
## Motivation

Due to the addition of parameters of Degree 16 to the SnarkVM binaries via the `impl_local` strategy, the wasm binary size increased to `17mb`. This PR changes all `wasm` parameter downloads to be downloaded remotely to optimize the size of the wasm binary. As a result, the wasm binary size drops to `7 mb`. This PR also adds missing wasm feature flag invocations needed to enforce that the parameters crate uses the `wasm` flag.

This PR also expands the number of features exported via the `snarkvm-wasm` crate to the full set needed to support program executions and deployments. Implementors of aleo `wasm` libraries should now be able to simply import the `snarkvm-wasm` crate as a single dependency.

## Related PRs

Currently there's no way to save parameters that have already been downloaded for later re-use. [A checked powers extension](https://github.com/AleoHQ/snarkVM/pull/1987/files#diff-23b90077c1b228bf00706c47481ee59e95b22d419d7ae1b71fd810a0fa6a9137R477) method in the  [Async Parameter Preloading PR](https://github.com/AleoHQ/snarkVM/pull/1987) proposes a safe way to use saved parameters by inserting them with several verification checks.

### CODE REVIEW

File: wasm/Cargo.toml [hunk 1]
```diff
@@ -31,6 +31,31 @@ path = "../console"
...
+features = [ "async", "wasm" ]
+optional = true
+
+[dependencies.snarkvm-ledger-store]
+path = "../ledger/store"
+version = "=0.15.4"
+features = [ "wasm" ]
+optional = true
+
+[dependencies.snarkvm-circuit-network]
```
Comment (howardwu) [inline wasm/Cargo.toml]: nit: order alphabetically

File: parameters/src/testnet3/mod.rs [hunk 2]
```diff
@@ -21,8 +21,14 @@ pub use powers::*;
 const REMOTE_URL: &str = "https://s3-us-west-1.amazonaws.com/testnet3.parameters";
 
 // Degrees
+#[cfg(not(feature = "wasm"))]
 impl_local!(Degree15, "resources/", "powers-of-beta-15", "usrs");
+#[cfg(not(feature = "wasm"))]
 impl_local!(Degree16, "resources/", "powers-of-beta-16", "usrs");
+#[cfg(feature = "wasm")]
+impl_remote!(Degree15, REMOTE_URL, "resources/", "powers-of-beta-15", "usrs");
```
Comment (howardwu) [inline parameters/src/testnet3/mod.rs]:
```suggestion
#[cfg(feature = "wasm")]
impl_remote!(Degree15, REMOTE_URL, "resources/", "powers-of-beta-15", "usrs");
#[cfg(not(feature = "wasm"))]
impl_local!(Degree16, "resources/", "powers-of-beta-16", "usrs");
```

File: parameters/src/testnet3/mod.rs [hunk 3]
```diff
@@ -37,7 +43,13 @@ impl_remote!(Degree27, REMOTE_URL, "resources/", "powers-of-beta-27", "usrs");
 impl_remote!(Degree28, REMOTE_URL, "resources/", "powers-of-beta-28", "usrs");
 
 // Shifted Degrees
+#[cfg(not(feature = "wasm"))]
 impl_local!(ShiftedDegree15, "resources/", "shifted-powers-of-beta-15", "usrs");
+#[cfg(not(feature = "wasm"))]
+impl_local!(ShiftedDegree16, "resources/", "shifted-powers-of-beta-16", "usrs");
+#[cfg(feature = "wasm")]
+impl_remote!(ShiftedDegree15, REMOTE_URL, "resources/", "shifted-powers-of-beta-15", "usrs");
```
Comment (howardwu) [inline parameters/src/testnet3/mod.rs]:
```suggestion
#[cfg(feature = "wasm")]
impl_remote!(ShiftedDegree15, REMOTE_URL, "resources/", "shifted-powers-of-beta-15", "usrs");
#[cfg(not(feature = "wasm"))]
impl_local!(ShiftedDegree16, "resources/", "shifted-powers-of-beta-16", "usrs");
```

File: wasm/Cargo.toml [hunk 4]
```diff
@@ -77,9 +102,11 @@ version = "0.3.37"
...
 [features]
 default = [ "full" ]
-full = [ "console", "curves", "fields", "synthesizer", "utilities" ]
-console = [ "snarkvm-console" ]
+full = [ "console", "curves", "fields", "ledger", "synthesizer", "utilities" ]
+console = [ "snarkvm-console/default" ]
 curves = [ "snarkvm-curves" ]
 fields = [ "snarkvm-fields" ]
+ledger = [ "snarkvm-ledger-block", "snarkvm-ledger-query", "snarkvm-ledger-store" ]
+network = [ "snarkvm-circuit-network", "snarkvm-console/network" ]
```
Comment (howardwu) [inline wasm/Cargo.toml]: nit: move this into `circuit` and `console`, and update `full` to include `circuit`.

**Review by howardwu — APPROVED** [review]
LGTM, pending changes.

**Review by raychu86 — APPROVED** [review]
lgtm

**Pull request merged**